test/schema.test.js

```diff
@@ -2430,4 +2430,33 @@ describe('schema', function() {
       assert.throws(buildInvalidSchema, /`db` may not be used as a schema pathname/);
     });
   });
+
+  describe('mongoose.set(`strictQuery`, value); (gh-6658)', function() {
+    let strictQueryOriginalValue;
+
+    this.beforeEach(() => strictQueryOriginalValue = mongoose.get('strictQuery'));
+    this.afterEach(() => mongoose.set('strictQuery', strictQueryOriginalValue));
```

Comment:
@vkarpov15 I'd like to know what you think of this approach.

I initially made the cleanup in each test, but if one of those tests had failed, the cleanup wouldn't have happened, causing more tests to fail, making it difficult to find the root cause of all the failing tests.

I think this mitigates this concern, but I still don't like it. How would you do that?

Reply:
This approach is ok, but the better approach would be to create a new Mongoose instance using `const _mongoose = new mongoose.Mongoose();` and use `_mongoose.set()`. Check out the test for #6578 for an example.

Reply:
Wouldn't this attempt to create a new connection with the same URI for the database? Making tests run significantly slower?